File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> hm = new HashMap<>();
        for (String s : strs) {
            char[] w = s.toCharArray();
            Arrays.sort(w);
            String sortedWord = new String(w);
            if (!hm.containsKey(sortedWord)) {
                hm.put(sortedWord, new ArrayList<>());
            }
            hm.get(sortedWord).add(s);
        }
        return new ArrayList<>(hm.values());
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        words = {}
        for s in strs:
            sortedWord = ''.join(sorted(s))
            if sortedWord not in words:
                words[sortedWord] = []
            words[sortedWord].append(s)
        return list(words.values())   